src/org/labkey/test/tests/TextChoiceImportExportAndOtherDomainsTest.java

```diff
@@ -0,0 +1,376 @@
+import org.labkey.test.params.FieldDefinition;
+import org.labkey.test.util.DataRegionTable;
+import org.labkey.test.util.IssuesHelper;
+import org.labkey.test.util.ListHelper;
+import org.labkey.test.util.PortalHelper;
+import org.labkey.test.util.SampleTypeHelper;
+import org.openqa.selenium.WebElement;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Category({Daily.class})
+public class TextChoiceImportExportAndOtherDomainsTest extends TextChoiceTest
+{
+
+    private static final String ORIGINAL_PROJ_NAME = "TextChoice_Import_Export_Test";
+    private static final String IMPORTED_PROJ_NAME = "Imported_TextChoice_Import_Export_Test";
+
+    private static Map<String, String> assayResultRowData = new HashMap<>();
+
+    // List field name, values etc...
+    private static final String LIST_NAME = "Simple_TC_List";
+    private static final String LIST_TC_FIELD = "LTC_Field";
+    private static final String LIST_TEXT_FIELD = "Str";
+    private static final List<String> LIST_VALUES = Arrays.asList("L1", "L2", "L3", "L4");
+    private static List<Map<String, String>> listData = new ArrayList<>();
+
+    // Issue names, etc...
+    private static final String ISSUE_DESIGN_NAME = "Simple_TC_Issue";
+    private static final String ISSUE_TC_FIELD = "Issue_TC_Field";
+    private static final List<String> ISSUE_FIELD_VALUES = List.of("I1", "I2", "I3", "I4");
+    private static final String ISSUE_TITLE = "Issue to validate TextChoice fields";
+    private static final String ISSUE_VALUE = ISSUE_FIELD_VALUES.get(2);
+
+    @Override
+    protected String getProjectName()
+    {
+        return ORIGINAL_PROJ_NAME;
+    }
+
+    @Override
+    protected void doCleanup(boolean afterTest) throws TestTimeoutException
+    {
+        _containerHelper.deleteProject(ORIGINAL_PROJ_NAME, afterTest);
+        _containerHelper.deleteProject(IMPORTED_PROJ_NAME, afterTest);
+    }
+
+    @BeforeClass
+    public static void setupProject()
+    {
+        TextChoiceImportExportAndOtherDomainsTest init = (TextChoiceImportExportAndOtherDomainsTest) getCurrentTest();
+        init.doSetup();
+    }
+
+    private void doSetup()
+    {
+
+        PortalHelper portalHelper = new PortalHelper(this);
+        _containerHelper.createProject(getProjectName(), null);
+        _containerHelper.enableModule("Issues");
+
+        log("Add some web parts to make it easier to debug etc...");
+        goToProjectHome();
+        portalHelper.enterAdminMode();
+        portalHelper.addWebPart("Sample Types");
+        portalHelper.addWebPart("Assay List");
+        portalHelper.addWebPart("Lists");
+        portalHelper.addWebPart("Issue Definitions");
+        portalHelper.addWebPart("Issues Summary");
+        portalHelper.exitAdminMode();
+
+    }
+
+    /**
+     * Create a run for the assay.
+     */
+    protected void createAssayRun()
```

Comment:
how similar is this to the `createAndValidateAssayRun` method in the TextChoiceAssayTest? any reason for looking to share the assay run import parts?

Reply:
Yep, had thought about that. Moved the create run code out to a shared method.